assessment_software/src/app/course-card/course-card.component.ts

```diff
@@ -0,0 +1,18 @@
+import { Component, Input, Output, EventEmitter } from '@angular/core';
+
+@Component({
+  selector: 'app-course-card',
+  templateUrl: './course-card.component.html',
+  styleUrls: ['./course-card.component.css']
+})
+export class CourseCardComponent {
+  @Input() courseName:string='Course Name';
+  @Input() year_semester= new Date().getFullYear() + "/1";
+  @Input() assignments:Number=0;
+  @Input() submissions:Number=0;
+  @Output() onCourseCardButtonClicked= new EventEmitter;
+
+  handleClick(event:Event){
+    this.onCourseCardButtonClicked.emit(event);
+  }
```

Comment:
We will need more callbacks in addition to the whole card being clicked.

Revise by adding this eventemitters: 
- onArchiveButtonClicked
- onAddAssignmentButtonClicked
- onAddStudentButtonClicked

This way, later when the card is being used when a user clicks on this button, we can load the necessary pages accordingly.